cypress/integration/students.spec.js

```diff
@@ -20,4 +20,64 @@ describe("Students", function() {
       });
   });
 });
-  
+
+describe("Updating Students", function() {
+  beforeEach(() => {
+    cy.task("taskTruncateTables");
+    cy.task("taskCreateStudent");
+  });
+  it("returns error for non-existant student", function() {
+    cy.request({
+      method: "PUT",
+      url: "api/students/0"
+    }).then(res => {
+      expect(res.body.status).to.eq("failure");
+      expect(res.body.errors).to.eq("Missing record");
+    });
+  });
+  it("returns error for invalid parameters", function() {
+    cy.request({
+      method: "GET",
+      url: "api/students"
+    }).then(res => {
+      expect(res.body.students[0].id).to.be.a("number");
+      cy.request({
+        method: "PUT",
+        url: `api/students/${res.body.students[0].id}`,
+        body: {invalid: "invalid"}
+      }).then(res => {
+        expect(res.body.status).to.eq("failure");
+        expect(res.body.errors).to.eq("Invalid params");
+      });});
+  });
+  it("returns error for blank input", function() {
+    cy.request({
+      method: "GET",
+      url: "api/students"
+    }).then(res => {
+      expect(res.body.students[0].id).to.be.a("number");
+      cy.request({
+        method: "PUT",
+        url: `api/students/${res.body.students[0].id}`,
+        body: {firstName: ""}
+      }).then(res => {
+        expect(res.body.status).to.eq("failure");
+        expect(res.body.errors).to.eq("No input");
+      });});
+  });
+  it("returns a successful update", function() {
+    cy.request({
+      method: "GET",
+      url: "api/students"
+    }).then(res => {
+      expect(res.body.students[0].id).to.be.a("number");
```

Comment:
Can we add an assertion on the name so we can see that it changes on a successful update?

Reply:
``
      cy.request({
      method: "PUT",
      url: "api/students/1",
      body: {firstName: "Fred"}
    }).then(res => {
      expect(res.body.status).to.eq("success");
      expect(res.body.student.firstName).to.eq("Fred");
```
Such as this or do you mean another test afterwards that gets api/students and tests for the new name?

Reply:
```suggestion
      cy.request({
      method: "PUT",
      url: "api/students/1",
      body: {firstName: "Fred"}
    }).then(res => {
      expect(res.body.status).to.eq("success");
      expect(res.body.student.firstName).to.eq("Fred");
```

Such as this or do you mean another test afterwards that gets api/students and tests for the new name?

Reply:
tldr - it doesn't matter anymore if you're making 1 put request.

If you're no longer making 2 requests - the first request to get all the students, it doesn't matter as much - we can rely on looking at the task for the previous name to verify the update actually happens. If you were getting data first before updating, this extra assertion would have added confidence that the update is working as expected (decreasing reliance on how the students endpoint works)